Try to make some FAQ questions more inviting

matta opened the pull request:
Replace the "why we don't want to discuss things" question with three
questions covering substantially the same topic. Try to strike a balance
between inviting participation and feedback while still making it clear
that the evolution process produces "firm" decisions.

Address issue #1833.

matta committed:
Try to make some FAQ questions more inviting

Replace the "why we don't want to discuss things" question with three
questions covering substantially the same topic. Try to strike a balance
between inviting participation and feedback while still making it clear
that the evolution process produces "firm" decisions.

jonmeow commented on docs/project/faq.md:
```diff
@@ -486,36 +488,59 @@ other parts of LLVM will be advantageous, but not required.
+to understand the community's reaction to Carbon.
+
+We encourage you to investigate why Carbon came to be the way it is. Look for
+proposals, open issues, and past discussions. Read through and understand the
```
```suggestion
We encourage you to investigate why Carbon came to be the way it is. Designs will include links to the proposals and important alternatives considered that led to them, typically linked at the bottom. Read through and understand the
```

Trying to explain _where_ to look. "open issues" I think typically shouldn't be a resource. Part of where I think things get lost, though, is _where_ to look -- so trying to directly capture that.

jonmeow commented on docs/project/faq.md:
```diff
@@ -486,36 +488,59 @@ other parts of LLVM will be advantageous, but not required.
+
+The Carbon language is still young, and there are aspects of the language that
+are provisional and expected to change.
+
```
```suggestion
```

This feels out of place here, I think it reads better without.

jonmeow commented on docs/project/faq.md:
```diff
@@ -486,36 +488,59 @@ other parts of LLVM will be advantageous, but not required.
+Both Discord and Github Discussions allow you to give an emoji "reaction" to
+individual posts. If you'd like to amplify what has already been said, please
+use these instead of posting messages that re-state substantially the same
+thing. These make conversations easier to follow, and make it easier to
```
```suggestion
thing. These make conversations easier to follow and
```

jonmeow commented on docs/project/faq.md:
```diff
@@ -486,36 +488,59 @@ other parts of LLVM will be advantageous, but not required.
+
+### What can I do if I disagree with a design decision?
+
+We invite you to give us constructive feedback. Some of Carbon's design
```
```suggestion
The Carbon language is still young, and there are aspects of the language that
are provisional and expected to change. We invite you to give us constructive feedback. Some of Carbon's design
```

Maybe more appropriate up here? I'm uncertain.

matta replied:
I ended up removing the "Carbon is still young" sentence since it didn't seem valuable in context.

jonmeow commented on docs/project/faq.md:
```diff
@@ -486,36 +488,59 @@ other parts of LLVM will be advantageous, but not required.
-problematic before pushing for change.
+See also the related questions
+[What happens when a decision was wrong?](#what-happens-when-a-decision-was-wrong),
+[How does Carbon make decisions?](#how-does-carbon-make-decisions) and
```
```suggestion
[How does Carbon make decisions?](#how-does-carbon-make-decisions), and
```

We're trying to stick with Oxford commas

jonmeow commented on docs/project/faq.md:
```diff
@@ -486,36 +488,59 @@ other parts of LLVM will be advantageous, but not required.
 specific tools we use when building.
 
-### Why aren't people willing to discuss certain topics, like the file extension or syntax for specific constructs?
+### When do we re-visit decisions or re-open discussions?
```
```suggestion
### When do we revisit decisions or reopen discussions?
```

Dashes aren't needed here

jonmeow commented on docs/project/faq.md:
```diff
@@ -486,36 +488,59 @@ other parts of LLVM will be advantageous, but not required.
+
+Changing decisions that have come out of the [evolution process](evolution.md)
+involves a formal process. See
+[When do we re-visit decisions or re-open discussions?](#when-do-we-re-visit-decisions-or-re-open-discussions).
```
```suggestion
[When do we revisit decisions or reopen discussions?](#when-do-we-revisit-decisions-or-reopen-discussions).
```

Because my other suggestion will break these

jonmeow reviewed (commented):
I agree this is an improvement, thank you. Making some suggestions on wording.

matta commented:
> I agree this is an improvement, thank you. Making some suggestions on wording.

Thanks for the suggested edits @jonmeow, I've accepted them all, deleted the "Carbon is still young" sentence, and fixed up presubmits. Ready for another look.

jonmeow reviewed (approved):
Thanks!

Just an FYI, but we avoid force-pushing because it confuses GitHub's comment system. We'll use GitHub's squash-and-merge feature, so it's not necessary to have one commit per PR.
